### PR TITLE
Implement color theming support

### PR DESCRIPTION
**Pull request by jordanjones243**
# Alaska Airlines Pull Request

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

**Resolves:** #198 

## Summary:

_Please summarize the scope of the changes you have submitted, what the intent of the work is and anything that describes the before/after state of the project._

* Update dependencies
* Implement custom versioned `auro-dropdown` component
* Implement tier 3 color tokens

## Type of change:

Please delete options that are not relevant.

- [x] New capability
- [x] Revision of an existing capability
- [ ] Infrastructure change (automation, etc.)
- [ ] Other (please elaborate)

## Checklist:

- [x] My update follows the CONTRIBUTING guidelines of this project
- [x] I have performed a self-review of my own update

**By submitting this Pull Request, I confirm that my contribution is made under the terms of the Apache 2.0 license.**

_Pull Requests will be evaluated by their quality of update and whether it is consistent with the goals and values of this project. Any submission is to be considered a conversation between the submitter and the maintainers of this project and may require changes to your submission._

**Thank you for your submission!**<br>
-- Auro Design System Team

## Summary by Sourcery

Implement color theming support by adding tier 3 color tokens and refactor the `auro-dropdown` component to use a custom versioned tag. Update documentation and tests to reflect these changes, and remove deprecated husky script.

New Features:
- Implement color theming support by introducing tier 3 color tokens and allowing components to be restyled using new SCSS variables.

Enhancements:
- Refactor the `auro-dropdown` component to use a custom versioned tag for better dependency management.

Documentation:
- Update documentation to include information on theme support and how to restyle components using new tokens.

Tests:
- Update tests to accommodate changes in the `auro-dropdown` component's query selector.

Chores:
- Remove deprecated husky script and update ESLint configuration to use a new module-based setup.

### CODE REVIEW

**Review by jason-capsule42 — CHANGES_REQUESTED** [review]
Screenshot 2024-08-18 at 5 23 18 PM
The selected value is not rendering correctly in the menu for me.

File: .husky/_/husky.sh [hunk 1]
```diff
@@ -0,0 +1,9 @@
+echo "husky - DEPRECATED
+
+Please remove the following two lines from $0:
+
+#!/usr/bin/env sh
+. \"\$(dirname -- \"\$0\")/_/husky.sh\"
+
+They WILL FAIL in v10.0.0
+"
```
Comment (jason-capsule42) [inline .husky/_/husky.sh:9]: Add the end of file extra line

File: src/color.scss [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+
+:host([disabled]) {
+  * {
+    color: var(--ds-color-text-disabled-default, $ds-color-text-disabled-default);
+  }
+}
+
+:host([validity]:not([validity='valid'])) {
+  .selectElement-helpText {
+    --ds-auro-select-help-text-color: var(--ds-color-text-error-default, #{$ds-color-text-error-default});
```
Comment (jason-capsule42) [inline src/color.scss:40]: Why does select have to do any work on the help text coloring at all? This should be handled by dropdown entirely.

**Review by jason-capsule42 — CHANGES_REQUESTED** [review]

File: src/color.scss (hunk 2 above)
Comment (jordanjones243) [reply src/color.scss:40]: That is a great point that we can discuss. Looking at the javascript code, it looks like auro-select is currently handling its own validation, which should be handled at the dropdown level.

Comment (jason-capsule42) [reply src/color.scss:40]: dropdown doesn't "do" validation. But it does render the results of the validation simply by adding/removing the "error" attribute to the dropdown element in the render function.

I would think you should be able to just remove all the select style code for validation and, verify the error attribute is passed to dropdown correctly, and then retest it.

**Comment by jason-capsule42** [comment]
@sourcery-ai review

**Review by jason-capsule42 — DISMISSED** [review]

**Comment by jordanjones243** [comment]
PR was approved with @jason-capsule42 during pair programming and I will therefore be forcing a release.

**Pull request merged**